cluster-controller/src/main/java/io/strimzi/controller/cluster/ClusterController.java

```diff
@@ -48,16 +42,20 @@
 
     private Watch configMapWatch;
 
-    private OperationExecutor opExec = null;
-
     private long reconcileTimer;
+    private final ZookeeperClusterOperation zookeeperClusterOperation;
+    private final KafkaClusterOperation kafkaClusterOperation;
+    private final KafkaConnectClusterOperation kafkaConnectClusterOperation;
 
-    public ClusterController(ClusterControllerConfig config) throws Exception {
+    public ClusterController(ClusterControllerConfig config) {
         log.info("Creating ClusterController");
 
         this.namespace = config.getNamespace();
         this.labels = config.getLabels();
         this.k8s = new K8SUtils(new DefaultKubernetesClient());
+        this.zookeeperClusterOperation = new ZookeeperClusterOperation(vertx, k8s);
+        this.kafkaClusterOperation = new KafkaClusterOperation(vertx, k8s);
+        this.kafkaConnectClusterOperation = new KafkaConnectClusterOperation(vertx, k8s);
```

Comment:
What is the benefit of creating the operations in the constructor instead of using singleton or something similar?

Reply:
By using mock `*ClusterOperations` I can unit test the `ClusterController` more easily. Using a singleton would couple the `ClusterController` quite tightly to the `*ClusterOperations`, which would tend to prevent such testing.